### PR TITLE
Time selector JS update

### PR DESCRIPTION
**Pull request by dandwal**
### What

- New JS for year grouped range option checker
- Improved JS speed (selectors are expensive) for date range updater.
- Added more margin to initial error message 
- Time selector now has additional JS to throw errors

### How to review

Check on CMD filter by time datasets (such as CPIH) that the ranges still work and correct errors are still thrown when appropriate.

Can be tested with the following PRs in conjunction:
- https://github.com/ONSdigital/dp-frontend-models/pull/73
- https://github.com/ONSdigital/dp-frontend-renderer/pull/513
- https://github.com/ONSdigital/dp-frontend-filter-dataset-controller/pull/233

### Who can review

Anyone except me

### CODE REVIEW

File: js/app/date-range-updater.js [hunk 1]
```diff
@@ -1,38 +1,78 @@
-$(document).ready(function() {
-    $("#start-month").change(function() {
+$(document).ready(function () {
```
Comment (rav-pradhan) [inline js/app/date-range-updater.js:1]: Currently when you return to this page to edit what you saved previously, the End Year contains all options, so you can then select a year that precedes the Start Year. Should we implement the logic that filters out illogical End Years to also run on page load if the user is returning to it?

**Review by rav-pradhan — COMMENTED** [review]
Mostly looks good. Can select months and also relevant years. Clicking Save and Return afterwards displays the relevant month/years in the filter overview page.

Just a couple small comments (one of which was an existing thing, just wanted to get your thoughts)

File: js/app/date-range-updater.js (hunk 1 above)
Comment (dandwal) [reply js/app/date-range-updater.js:1]: Ha, that is a very good catch, looks like a bug we have in the Range selection too.

File: js/app/time-selector.js [hunk 2]
```diff
@@ -168,33 +168,62 @@ $(document).ready(function() {
...
+                    if ($("#grouped-range-error-message").length === 0) {
+                        groupedRange.prepend("<div id=\"grouped-range-error-message\" class=\"margin-bottom--1 font-size--16 form-error\"><strong>Select a range of years</strong></div>");
+                    }
+                }
+                if (errorList.length) {
+                    // Always go to the first error on form
                     $('html, body').animate({
-                        scrollTop: $("#multiple-choice-content-list").offset().top
+                        scrollTop: $(errorList[0]).offset().top
                     }, 2000);
```
[the remark this answers is not in the document]
Comment (dandwal) [reply js/app/time-selector.js:203]: I can see in other scroll areas of the website we have used 1000 instead, so I have matched this now so it should be one second.

**Review by rav-pradhan — APPROVED** [review]
Tested again and edge case fix looks good, also I do think it's better having the shorter scroll time 👍🏽

**Pull request merged**